src/devices_models/devices/common/timeseries_constraint.jl

```diff
@@ -22,20 +22,43 @@ Constructs upper bound for given variable and time series data and a multiplier.
 """
 function device_timeseries_ub(psi_container::PSIContainer,
                               ts_data::Vector{Tuple{String, Int64, Float64, Vector{Float64}}},
+                              range_data::DeviceRange, #TODO: there is some repititve info in range_data and ts_data, and consistend order/size are assumed, we should combine into a single argument
```

Comment:
a ts_data object hasn't been implemented yet in the same way that there is a device_range object. Eventually, when we resolve #219 this will be more clear.

There shouldn't be any changes to this function yet.

Reply:
Agreed on the need for #219 . As for the changes to this function, they support the desired features.  I suggest that we keep these changes and resolve the remainder of #219 in a new PR.